packages/core/src/exposed-thing.ts

```diff
@@ -110,15 +111,18 @@ export default class ExposedThing extends ConsumedThing implements TD.Thing, WoT
      */
     public writeProperty(propertyName: string, newValue: any): Promise<void> {
         return new Promise<void>((resolve, reject) => {
-            let state = this.interactionStates.get(propertyName);
+            let state = this.propertyStates.get(propertyName);
             if (state) {
-                let oldValue = state.value;
-                state.value = newValue;
-
-                // calls all handlers
-                state.handlers.forEach(handler => handler.apply(this, [newValue, oldValue]))
+                // call write handler (if any)
+                if (state.writeHandler != null) {
+                    console.log(`ExposedThing '${this.name}' calls registered writeHandler for property ${propertyName}`);
+                    state.value = state.writeHandler.apply(state.that, [newValue]); 
+                } else {
+                    console.log(`ExposedThing '${this.name}' sets new value ${newValue} for property ${propertyName}`);
+                    state.value = newValue;
+                }
 
-                resolve(newValue);
+                resolve(state.value);
```

Comment:
Do not return the written value
* cf. `Promise<void>` in signature
* cf. 2.04 Changed